hack/gen-install-calico-windows-script/main.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"text/template"
+)
+
+type install struct {
+	Product     string `json:"product"`
+	ProductName string `json:"productName"`
+	Version     string `json:"version"`
+	BaseUrl     string `json:"baseUrl"`
+	RootDir     string `json:"rootDir"`
+	ZipFileName string `json:"zipFileName"`
+}
+
+func newInstall(product, version, baseUrl string) (install, error) {
+	data := install{
+		Product: product,
+		Version: version,
+		BaseUrl: baseUrl,
+	}
+	var productName, rootDir, zipFileName string
+
+	if product == "Calico" {
+		productName = "Calico for Windows"
+		rootDir = "CalicoWindows"
+		zipFileName = "calico-windows.zip"
+	} else if product == "Calico Enterprise" {
+		productName = "Tigera Calico for Windows"
+		rootDir = "TigeraCalico"
+		zipFileName = "tigera-calico-windows.zip"
+	} else {
+		return data, fmt.Errorf("invalid product: %v", product)
+	}
+
+	data.ProductName = productName
+	data.RootDir = rootDir
+	data.ZipFileName = zipFileName
+	return data, nil
```

Comment:
Instead of having a function to assembly a install structure. I would suggest you to simplify it by:
- Define a map (global var) from product to install structure. Then it is easier to know those strings are global variable in case we need to change it later or make sure it is consistent across directories/repo.
```
var installations map[string]install {
     "Calico": install {
         Product: "Calico",
         ProductName: "Calico for Windows",
         RootDir: "CalicoWindows",
         ZipFileName: "calico-windows.zip",
     }, 
     "Calient": install{
         xxx
     },
}
```
So the main() function just get the install from the map and populate baseURL and version.